File: 50-powx-n/50-powx-n.c
bool isOdd(int x){
    return x % 2 != 0;
};

double myPow(double x, long long n){
    if (n == 0){
        return 1.0;
    };
    if (n < 0){
        return 1.0 / myPow(x,-1*n);
    };
    
    if(isOdd(n)){
        return x * myPow(x*x,(n-1)/2);
    }else{
        return myPow(x*x,n/2);
    };
};

File: 50-powx-n/50-powx-n.py
class Solution:
    
    
    
    def myPow(self, x: float, n: int) -> float:
        
        if n == 0:
            return 1
        
        if n < 0:
            return 1 / self.myPow(x,-n)
        
        isOdd = lambda x: x % 2 != 0
        
        if isOdd(n):
            
            return x * self.myPow(x*x, (n-1) // 2)
        
        else:
            
            return self.myPow(x*x, n // 2)
        
        
        
        
        